examples/usage/inventory.rs

```diff
@@ -0,0 +1,198 @@
+//! Display the Inventory of a specific Player in a Grid
+//! using a custom Relationship to associate items (Entities)
+//! with a Player
+use bevy::{color::palettes::tailwind::*, ecs::spawn::SpawnIter, prelude::*};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+fn main() -> AppExit {
+    App::new()
+        .insert_resource(ClearColor(SLATE_950.into()))
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, (startup, show_inventory).chain())
+        .add_observer(update_button_color_on::<Over>(SLATE_400.into()))
+        .add_observer(update_button_color_on::<Out>(SLATE_500.into()))
+        .add_observer(update_button_color_on::<Pressed>(GREEN_400.into()))
+        .add_observer(update_button_color_on::<Released>(SLATE_500.into()))
+        .run()
+}
+
+fn update_button_color_on<E>(
+    color: Color,
+) -> impl Fn(Trigger<Pointer<E>>, Query<&mut BackgroundColor, With<Button>>)
+where
+    E: std::fmt::Debug + Reflect + Clone,
+{
+    // An observer closure that accepts a Pointer Event and a Color. We do this to avoid needing to write four
+    // versions of this observer, each triggered by a different event and with a different hardcoded
+    // Event/Color. Instead, the event type is a generic, and the Event/Color is passed in.
+    move |trigger: Trigger<Pointer<E>>, mut buttons: Query<&mut BackgroundColor, With<Button>>| {
+        let Ok(mut background_color) = buttons.get_mut(trigger.target()) else {
+            return;
+        };
```

Comment:
Nit/food for thoughts: We may consider returning a `Result` ? using `?` would help reducing noise. but I'm not a fan of the default behaviour of panicking if a result is panicking.

Reply:
- @extrawurst had the same kind of boilerplate for https://github.com/bevyengine/bevy/pull/19245 ; some coordination on implementation may be fruitful.